spring-cloud-dataflow-tasklauncher/spring-cloud-dataflow-tasklauncher-function/src/main/java/org/springframework/cloud/dataflow/tasklauncher/TaskLauncherFunction.java

```diff
@@ -43,34 +43,30 @@
  *
  * @author David Turanski
  **/
-public class TaskLauncherFunction implements Function<LaunchRequest, Optional<LaunchResponse>>, InitializingBean {
-	private static final Log log = LogFactory.getLog(TaskLauncherFunction.class);
+public class TaskLauncherFunction implements Consumer<LaunchRequest>, InitializingBean {
 
-	static final String TASK_PLATFORM_NAME = "spring.cloud.dataflow.task.platformName";
+	private static final Logger log = LoggerFactory.getLogger(TaskLauncherFunction.class);
 
-	private final TaskOperations taskOperations;
+	public static final String TASK_PLATFORM_NAME = "spring.cloud.dataflow.task.platformName";
 
+	private final TaskOperations taskOperations;
 	private String platformName = "default";
 
 	public TaskLauncherFunction(TaskOperations taskOperations) {
 		Assert.notNull(taskOperations, "`taskOperations` cannot be null.");
 		this.taskOperations = taskOperations;
 	}
 
-	/**
-	 *
-	 * @param launchRequest the task launch request for the Data Flow server.
-	 * @return an {@code Optional<Long>} containing the task Id if the request is accepted or
-	 * empty otherwise.
-	 */
 	@Override
-	public Optional<LaunchResponse> apply(LaunchRequest launchRequest) {
+	public void accept(LaunchRequest request) {
```

Comment:
It looks like the return value was being ignored previously by the old consumer, so this void return should be fine.

Reply:
Change from Function to Consumer

Reply:
Right, at first I was concerned because the lack of return value but the code that uses this before did nothing w/ the return value so this should be fine.